servers/src/test/java/tachyon/master/next/filesystem/meta/InodeDirectoryTests.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.next.filesystem.meta;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.master.block.BlockId;
+
+/**
+ * Unit tests for tachyon.InodeDirectory
+ */
+public final class InodeDirectoryTests {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  @Test
+  public void addChildrenTest() {
+    InodeDirectory inodeDirectory = createInodeDirectory();
+    InodeFile inodeFile1 = createInodeFile(2);
+    InodeFile inodeFile2 = createInodeFile(3);
+    inodeDirectory.addChild(inodeFile1);
+    inodeDirectory.addChild(inodeFile2);
+    Assert.assertEquals(createBlockId(2), (long) inodeDirectory.getChildrenIds().get(0));
+    Assert.assertEquals(createBlockId(3), (long) inodeDirectory.getChildrenIds().get(1));
+  }
+
+  @Test
+  public void batchRemoveChildTest() {
+    InodeDirectory inodeDirectory = createInodeDirectory();
+    InodeFile inodeFile1 = createInodeFile(1);
+    InodeFile inodeFile2 = createInodeFile(2);
+    InodeFile inodeFile3 = createInodeFile(3);
+    inodeDirectory.addChild(inodeFile1);
+    inodeDirectory.addChild(inodeFile2);
+    inodeDirectory.addChild(inodeFile3);
+    Assert.assertEquals(3, inodeDirectory.getNumberOfChildren());
+    inodeDirectory.removeChild("testFile1");
+    Assert.assertEquals(2, inodeDirectory.getNumberOfChildren());
+    Assert.assertFalse(inodeDirectory.getChildrenIds().contains(createBlockId(1)));
+  }
+
+  @Test
+  public void equalsTest() {
+    InodeDirectory inode1 = new InodeDirectory("test1", 1, 0, System.currentTimeMillis());
+    InodeDirectory inode2 = new InodeDirectory("test2", 1, 0, System.currentTimeMillis());
+    Assert.assertTrue(inode1.equals(inode2));
+  }
+
+  @Test
+  public void getIdTest() {
+    Assert.assertEquals(1, createInodeDirectory().getId());
```

Comment:
I don't think so. All directory instances created independently, so I wouldn't expect it.